src/pykeen/models/base.py

```diff
@@ -1195,3 +1237,183 @@ def _can_slice(fn) -> bool:
+        predict_with_sigmoid: bool = False,
+        automatic_memory_optimization: Optional[bool] = None,
+        preferred_device: DeviceHint = None,
+        random_seed: Optional[int] = None,
+        regularizer: Optional[Regularizer] = None,
+        entity_initializer: Optional[Initializer] = None,
+        entity_initializer_kwargs: Optional[Mapping[str, Any]] = None,
+        entity_normalizer: Optional[Normalizer] = None,
+        entity_normalizer_kwargs: Optional[Mapping[str, Any]] = None,
+        entity_constrainer: Optional[Constrainer] = None,
+        entity_constrainer_kwargs: Optional[Mapping[str, Any]] = None,
+        relation_initializer: Optional[Initializer] = None,
+        relation_initializer_kwargs: Optional[Mapping[str, Any]] = None,
+        relation_normalizer: Optional[Normalizer] = None,
+        relation_normalizer_kwargs: Optional[Mapping[str, Any]] = None,
+        relation_constrainer: Optional[Constrainer] = None,
+        relation_constrainer_kwargs: Optional[Mapping[str, Any]] = None,
+    ) -> None:
+        super().__init__(
+            triples_factory=triples_factory,
+            embedding_dim=embedding_dim,
+            relation_dim=relation_dim,
+            automatic_memory_optimization=automatic_memory_optimization,
+            loss=loss,
+            predict_with_sigmoid=predict_with_sigmoid,
+            preferred_device=preferred_device,
+            random_seed=random_seed,
+            regularizer=regularizer,
+            entity_initializer=entity_initializer,
+            entity_initializer_kwargs=entity_initializer_kwargs,
+            entity_normalizer=entity_normalizer,
+            entity_normalizer_kwargs=entity_normalizer_kwargs,
+            entity_constrainer=entity_constrainer,
+            entity_constrainer_kwargs=entity_constrainer_kwargs,
+            relation_initializer=relation_initializer,
+            relation_initializer_kwargs=relation_initializer_kwargs,
+            relation_normalizer=relation_normalizer,
+            relation_normalizer_kwargs=relation_normalizer_kwargs,
+            relation_constrainer=relation_constrainer,
+            relation_constrainer_kwargs=relation_constrainer_kwargs,
+        )
+        self.interaction_function = interaction_function
+
+        # extra embeddings
+        self.reverse_entity_embeddings = Embedding.init_with_device(
+            num_embeddings=triples_factory.num_entities,
+            embedding_dim=embedding_dim,
+            device=self.device,
+        )
+        self.reverse_relation_embeddings = Embedding.init_with_device(
+            num_embeddings=triples_factory.num_relations,
+            embedding_dim=embedding_dim,
+            device=self.device,
+        )
+
+    def forward(
+        self,
+        h_indices: Optional[torch.LongTensor],
+        r_indices: Optional[torch.LongTensor],
+        t_indices: Optional[torch.LongTensor],
+    ) -> torch.FloatTensor:  # noqa: D102
+        return 0.5 * sum(
```

Comment:
this looks a lot like an average....

why stop at 2 sided model, why not have an n-sided model?

Reply:
Because it is once with the triple, and once with the flipped triple. There is not natural third way of flipping :wink:

Reply:
It started from SimplE, which essentially is a two-sided model with DistMult base-model.